dashboard/15-final/app/ui/invoices/table-search.tsx

```diff
@@ -0,0 +1,69 @@
+'use client';
+
+import { MagnifyingGlassIcon } from '@heroicons/react/24/outline';
+import { usePathname, useRouter } from 'next/navigation';
+import { useTransition } from 'react';
+
+export default function TableSearch({ disabled }: { disabled?: boolean }) {
+  const { replace } = useRouter();
+  const pathname = usePathname();
+  const [isPending, startTransition] = useTransition();
+
+  function handleSearch(term: string) {
+    const params = new URLSearchParams(window.location.search);
```

Comment:
I think we can `useSearchParams` instead of `window.location.search`. This way, the next router becomes the source of truth. 

Here's an example: https://github.com/vercel/app-playground/blob/48042c4211bd672b0e5dd71bf0e0ad17d2429d94/app/snippets/search-params/client.tsx#L22